### PR TITLE
Enable StyleCop rules 1205, 1411, 1026, and 1000

### PR DESCRIPTION
**Pull request by stephentoub**
Experimenting with a single .sln for all of corefx, which makes it easier to apply such changes via auto-fixes.

### CODE REVIEW

File: src/System.Drawing.Common/src/misc/DbgUtil.cs [hunk 1]
```diff
@@ -17,7 +17,7 @@ internal sealed class DbgUtil
         public static void AssertFinalization(object obj, bool disposing)
         {
 #if GDI_FINALIZATION_WATCH
-            if( disposing || AppDomain.CurrentDomain.IsFinalizingForUnload() )
+            if ( disposing || AppDomain.CurrentDomain.IsFinalizingForUnload() )
```
Comment (safern) [inline src/System.Drawing.Common/src/misc/DbgUtil.cs]: Is there a rule for spaces in between the `(` `)` and the closest token inside them?

Comment (stephentoub) [reply src/System.Drawing.Common/src/misc/DbgUtil.cs]: There is.  There's a rule for pretty much everything :)

Comment (safern) [reply src/System.Drawing.Common/src/misc/DbgUtil.cs]: 😄 -- cool, cause now that I saw those lines, they look bad.

How hard was to create the huge .sln and auto-apply the code fixes?

File: src/Common/src/CoreLib/System/Decimal.DecCalc.cs [hunk 2]
```diff
@@ -690,7 +690,7 @@ private static unsafe int ScaleResult(Buf24* bufRes, uint hiRes, int scale)
                         }
 
                         break;
-                    } // for(;;)
+                    } // for (;;)
```
Comment (ViktorHofer) [inline src/Common/src/CoreLib/System/Decimal.DecCalc.cs]: I'm surprised that commented out instructions are also captured.

Comment (stephentoub) [reply src/Common/src/CoreLib/System/Decimal.DecCalc.cs]: > I'm surprised that commented out instructions are also captured.

They're not.  I also did a search/replace after the fact to catch stragglers like this.

**Review by ViktorHofer — APPROVED** [review]
Spot checked again.

**Review by safern — APPROVED** [review]

File: src/System.Drawing.Common/src/misc/DbgUtil.cs (hunk 1 above)
Comment (stephentoub) [reply src/System.Drawing.Common/src/misc/DbgUtil.cs]: > cool, cause now that I saw those lines, they look bad.

:smile:

> How hard was to create the huge .sln and auto-apply the code fixes?

Creating a hacky .sln was pretty easy.  I could check it in if desirable, though I expect it'll get out of date relatively quickly.  The less ideal part is it's a super gigantic .sln, and it quickly hits up against the limits of VS's "fix all" functionality.  @sharwell helped me workaround some of them.  In other cases I hit issues that I manually worked around.

Comment (safern) [reply src/System.Drawing.Common/src/misc/DbgUtil.cs]: I see. Is it possible to do this through a Console App, that it opens every solution in corefx and auto-applies the code fixes in all of them?

Comment (stephentoub) [reply src/System.Drawing.Common/src/misc/DbgUtil.cs]: > Is it possible to do this through a Console App, that it opens every solution in corefx and auto-applies the code fixes in all of them?

The dotnet-format tool is supposed to do that, at least eventually, but I don't think it's "there" yet.  @sharwell can speak to that.

**Pull request merged**